v2/piv/piv.go

```diff
@@ -664,6 +664,47 @@ func ykChangePUK(tx *scTx, oldPUK, newPUK string) error {
 	return err
 }
 
+// SetRetries sets the allowed retry count for the PIN and the PUK.
+//
+// Yubikeys allows one byte for storing each, allowed values are 1-255. In instances of greater
+// than 15 retries remaining, the remaining count will show 15 as Yubikeys only have 4 bits in
+// the response for remaining retries.
+//
+// IMPORTANT NOTE: Changing the retries on Yubikeys RESETS THE PIN AND PUK TO THEIR DEFAULTS!
+// If you use SetRetries, it is *highly* recommended that you follow it with SetPIN and SetPUK.
+// https://docs.yubico.com/yesdk/users-manual/application-piv/commands.html#set-pin-retries
+//
+//	if err := yk.SetRetries(piv.DefaultManagementKey, piv.DefaultPIN, 5, 4); err != nil {
+//		// ...
+//	}
+func (yk *YubiKey) SetRetries(managementKey []byte, pin string, pinRetries byte, pukRetries byte) error {
```

Comment:
Can this be an int rather than bytes?

```
func (yk *YubiKey) SetRetries(managementKey []byte, pin string, pinRetries, pukRetries int) error {
```

That matches the existing API signatures https://pkg.go.dev/github.com/go-piv/piv-go/v2/piv#YubiKey.Retries

Reply:
I had it accepting bytes because that is the space the number of retries is stored in, so it automatically handled my bounds... except that zero isn't allowed, so I already had to validate the lower bound of the input range anyway. I changed it to take int, added upper bound validation, and added a test for erroring on exceeding upper bound.